tap_square/client.py

```diff
@@ -220,93 +247,89 @@ def get_payments(self, start_time, bookmarked_cursor):
             body,
             'payments')
 
-    def get_roles(self, bookmarked_cursor):
-        headers = {
-            'Authorization': 'Bearer ' + self._access_token,
-            'Content-Type': 'application/json'
-        }
-        params = {}
-        url = 'https://connect.squareup.com/v1/me/roles'
-
+    def get_cash_drawer_shifts(self, location_id, start_time, bookmarked_cursor):
         if bookmarked_cursor:
-            params['batch_token'] = bookmarked_cursor
-
-        with singer.http_request_timer('GET roles'):
-            result = requests.get(url, headers=headers, params=params)
-
-        if result.status_code != 200:
-            raise RuntimeError(result.reason)
-
-        batch_token = get_batch_token_from_headers(result.headers)
-
-        yield (result.json(), batch_token)
-
-        while batch_token:
-            params['batch_token'] = batch_token
-            with singer.http_request_timer('GET roles'):
-                result = requests.get(url, headers=headers, params=params)
-
-            if result.status_code != 200:
-                raise RuntimeError(result.reason)
-
-            batch_token = get_batch_token_from_headers(result.headers)
-
-            yield (result.json(), batch_token)
+            cursor = bookmarked_cursor
+        else:
+            cursor = '__initial__' # initial value so while loop is always entered one time
 
-    def get_cash_drawer_shifts(self, location_id, start_time, bookmarked_cursor):
         end_time = utils.strftime(utils.now(), utils.DATETIME_PARSE)
-        with singer.http_request_timer('GET cash drawer shifts'):
-            result = self._client.cash_drawers.list_cash_drawer_shifts(
-                location_id=location_id,
-                begin_time=start_time,
-                end_time=end_time,
-                cursor=bookmarked_cursor,
-                limit=1000,
-            )
-
-        if result.is_error():
-            raise RuntimeError(result.errors)
-
-        yield (result.body.get('items', []), result.body.get('cursor'))
+        while cursor:
+            if cursor == '__initial__':
+                cursor = bookmarked_cursor
```

Comment:
Do we need this if block?

If we have a `bookmarked_cursor` passed in, then this gets skipped. If we do set this to `__initial__` above, then we hit this block and set `cursor = None` (or some other falsey value). But then we do nothing with `cursor` until we maybe overwrite it with `result.body.get('cursor')`, which could set it to `None` anyway.

Reply:
In the latest version I added a comment "# initial value so while loop is always entered one time". It's set to something to force the while loop to go through at least once.

Reply:
Fixed it, removed that block